Core/GDCore/Extensions/Builtin/CommonInstructionsExtension.cpp

```diff
@@ -118,6 +119,13 @@ BuiltinExtensionsImplementer::ImplementsCommonInstructionsExtension(
                      "res/foreach.png",
                      std::make_shared<gd::ForEachEvent>());
 
+  extension.AddEvent("ForEachStructure",
```

Comment:
I'm inclined to call this `ForEachChildVariable`.

The term "Variable children" allows to extend this to arrays later.